scrapers/DEU/DE-BY/index.js

```diff
@@ -0,0 +1,56 @@
+import * as fetch from '../../../lib/fetch.js';
+import * as datetime from '../../../lib/datetime.js';
+
+// Set county to this if you only have state data, but this isn't the entire state
+// const UNASSIGNED = '(unassigned)';
+
+const scraper = {
+  country: 'DEU',
+  state: 'DE-BY', // ISO 3166 notation
+  url: 'https://raw.githubusercontent.com/jgehrcke/covid-19-germany-gae/master/data.csv',
+  type: 'json',
+  timeseries: true,
+  curators: [
+    {
+      name: 'Dr. Jan-Philip Gehrcke',
+      email: 'jgehrcke@googlemail.com'
+    }
+  ],
+  sources: [
+    {
+      name: 'Robert Koch-Institut, Bavarian health ministry',
+      description: 'Fresh data obtained from Bavarian health ministry by ZEIT ONLINE'
+    }
+  ],
+  async scraper() {
+    const queryDayString = datetime.getYYYYMD(new Date(process.env.SCRAPE_DATE));
+    const data = await fetch.csv('this.url');
+
+    let queryDayCases = 0;
+    let queryDayDeaths = 0;
+
+    if (process.env.SCRAPE_DATE) {
+      // Get historical data for a specific day.
+      for (const row of data) {
+        const sampleDayString = datetime.getYYYYMD(new Date(row.time_iso8601));
+        if (sampleDayString === queryDayString) {
+          queryDayCases = row['DE-BY_cases'];
+          queryDayDeaths = row['DE-BY_deaths'];
+        }
+      }
+    } else {
+      // Rely on last entry in time series to be rather fresh.
```

Comment:
This makes sense. We've done this with the rest of them. However, if the scrape date is matched, and the requested date is not in the timeseries, we should also use the latest to avoid errors. It will be backfilled when the timeseries is updated, so it's OK.

See ITA and BRA, they both do this.

Reply:
Not sure I understand 

>  if the scrape date is matched

Pseudo code as in

`if SCRAPE_DATE AND SCRAPE_DATE not in dataset THEN return newest datapoint`?

That would probably lead to artifacts for scrape dates too far in the past. mhm, but I probably misunderstood!

Reply:
so it goes like this:

```
latestData = dataset(LATEST_DATE)
if LATEST_DATE is before SCRAPE_DATE
  angryErrorMessage
  use(latestData)
else
  if datasetContains(SCRAPE_DATE)
    use dataset(SCRAPE_DATE)
```

This might be the best example, it errors out when data is missing: https://github.com/lazd/coronadatascraper/blob/master/scrapers/CHE/zurich.js#L15-L28

This one doesn't check that the data is there: https://github.com/lazd/coronadatascraper/blob/master/scrapers/ITA/index.js#L14-L24

Neither does this one: https://github.com/lazd/coronadatascraper/blob/master/scrapers/BRA/index.js#L1104-L1114

Reply:
Okay, there are three special cases we'd like to handle consciously:
- two special cases where queryDate is before or after the time/ interval covered by the timeseries.
- special case where time series has a hole.

Patch coming.